rosidl_default_generators/CMakeLists.txt

```diff
@@ -5,13 +5,11 @@ project(rosidl_default_generators NONE)
 find_package(ament_cmake REQUIRED)
 
 ament_export_dependencies(ament_cmake_core)
-ament_export_dependencies(rosidl_cmake)
+ament_export_dependencies(rosidl_core_generators)
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
 endif()
 
-ament_package(
-  CONFIG_EXTRAS "rosidl_default_generators-extras.cmake.in"
```

Comment:
I did a test, and it appears the cmake variables are set transitively. I.e. depending on rosidl_default_generators continues to set the variable `rosidl_default_generators_INCLUDE_DIRS`, for example.

Reply:
that's pretty weird, did you clean your `install` folder first?
If not, the old config file will be there.

Reply:
needs confirmation, otherwise the PR looks good to me

Reply:
I think it works because of the `ament_export_dependencies` macros in the CMakeLists.txt. I noticed that the variables were **not** being set when we `find_package(rosidl_default_runtime)`; adding `ament_export_dependencies` fixed it ([07dd659](https://github.com/ros2/rosidl_defaults/pull/22/commits/07dd659ceaeb8889ab7cb954675565e7f17621e9))

Tested with the following cmake code after a fresh build (removed the build and install directories):

```cmake
find_package(rosidl_default_generators REQUIRED)

message("default-definitions: ${rosidl_default_generators_DEFINITIONS}")
message("default-include_dirs: ${rosidl_default_generators_INCLUDE_DIRS}")
message("default-libraries: ${rosidl_default_generators_LIBRARIES}")
message("core-definitions: ${rosidl_core_generators_DEFINITIONS}")
message("core-include_dirs: ${rosidl_core_generators_INCLUDE_DIRS}")
message("core-libraries: ${rosidl_core_generators_LIBRARIES}")
message("")

find_package(rosidl_default_runtime REQUIRED)
message("default-runtime-definitions: ${rosidl_default_runtime_DEFINITIONS}")
message("default-runtime-include_dirs: ${rosidl_default_runtime_INCLUDE_DIRS}")
message("default-runtime-libraries: ${rosidl_default_runtime_LIBRARIES}")
message("core-runtime-definitions: ${rosidl_core_runtime_DEFINITIONS}")
message("core-runtime-include_dirs: ${rosidl_core_runtime_INCLUDE_DIRS}")
message("core-runtime-libraries: ${rosidl_core_runtime_LIBRARIES}")
```

The `*_DEFINITIONS` variables are never set (but I don't think that has anything to do with this change.

Reply:
Nice, LGTM then